src/conv_transpose2d.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+import {Tensor} from './lib/tensor.js';
+import {validateConv2dParams} from './lib/validate-input.js';
+import {computePaddingForAutoPad} from './lib/compute-padding.js';
+import {transpose} from './transpose.js';
+
+/**
+ * Compute a 2-D transposed convolution given 4-D input and filter tensors.
+ * @param {Tensor} input
+ * @param {Tensor} filter
+ * @param {MLConvTranspose2dOptions} options
+ * @return {Tensor}
+ */
+export function convTranspose2d(input, filter, {padding = [0, 0, 0, 0],
+  strides = [1, 1],
+  groups = 1,
+  dilations = [1, 1],
+  outputPadding = [0, 0],
+  outputSizes,
+  activation = (x) => x,
+  inputLayout = 'nchw',
+  filterLayout = 'iohw',
+  bias,
+  autoPad = 'explicit',
+}
+= {}) {
+  if (inputLayout === 'nhwc') {
+    // nhwc -> nchw
+    input = transpose(input, {permutation: [0, 3, 1, 2]});
+  }
+  if (filterLayout === 'iohw') {
+    // iohw -> oihw
+    filter = transpose(filter, {permutation: [1, 0, 2, 3]});
+  } else if (filterLayout === 'hwoi') {
+    // hwoi -> oihw
+    filter = transpose(filter, {permutation: [2, 3, 0, 1]});
+  } else if (filterLayout === 'ohwi') {
+    // ohwi -> oihw
+    filter = transpose(filter, {permutation: [0, 3, 1, 2]});
+  }
+
+  validateConv2dParams(input, filter, {groups, bias});
+
+  const [batchCount, inputChannels, inputHeight, inputWidth] = input.shape;
+  const [outputChannels, , filterHeight, filterWidth] = filter.shape;
+  const [strideHeight, strideWidth] = strides;
+  const [dilationHeight, dilationWidth] = dilations;
+  const effectiveFilterHeight = filterHeight + (filterHeight - 1) * (dilationHeight - 1);
+  const effectiveFilterWidth = filterWidth + (filterWidth - 1) * (dilationWidth - 1);
```

Comment:
I've more commonly seen the simpler equation used:

`const effectiveFilterWidth = (filterWidth - 1) * dilationWidth + 1;`

e.g. width = 5, dilation = 3
`5 + (5 - 1) * (3 - 1) = 5 + 4 * 2 = 13`
`(5 - 1) * 3 + 1 = 4 * 3 + 1 = 13`

Reply:
> const effectiveFilterWidth = (filterWidth - 1) * dilationWidth + 1;

Yes, it is more imaginable. 

Current WebNN API Spec described [convTranspose2d](https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-conv2d): 

> output size = 1 + (input size - filter size - (filter size - 1) * (dilation - 1) + beginning padding + ending padding) / stride

Maybe we need update to
> output size = 1 + (input size - (filter size - 1) * dilation - 1 + beginning padding + ending padding) / stride

same to compute the [`output size` of `conv2d`](https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-conv2d), @huningxin WDYT? Thanks.

Reply:
I've submitted https://github.com/webmachinelearning/webnn/pull/328 to WebML CG for getting feedbacks.